packages/widget.vite-plugin/src/plugin.ts

```diff
@@ -235,57 +257,52 @@ export function FoundryWidgetVitePlugin(options: Options = {}): Plugin {
               return;
             }
 
-            // TODO: Actually handle the widget RID from within the config, which will require somehow parsing the config
-            // Unfortunately, moduleParsed is not called during vite's dev mode for performance reasons, so the config file
-            // will need to be parsed/read a different way
-            fetch(
-              `${url.origin}/view-registry/api/dev-mode/${foundryConfig.widget.rid}/settings`,
-              {
-                body: JSON.stringify({
-                  entrypointJs: [
-                    ...entrypointToJsSourceFileMap[entrypointFileName],
-                  ].map((file) => ({
-                    scriptType: {
-                      type: "module",
-                      module: {},
-                    },
-                    filePath: `${localhostUrl}${file}`,
-                  })),
-                  entrypointCss: [],
-                }),
-                method: "POST",
-                headers: {
-                  authorization: `Bearer ${process.env.FOUNDRY_TOKEN}`,
-                  accept: "application/json",
-                  "content-type": "application/json",
-                },
-              },
-            )
-              .then((devResponse) => {
-                if (devResponse.status === 200 || devResponse.status === 204) {
-                  res.setHeader("Content-Type", "application/json");
-                  res.end(
-                    JSON.stringify({
-                      redirectUrl:
-                        `${url.origin}/workspace/custom-views/preview/${foundryConfig.widget.rid}`,
-                    }),
-                  );
-                } else {
-                  res.statusCode = devResponse.status;
-                  res.statusMessage =
-                    `Unable to start dev mode in Foundry (see terminal for more): ${devResponse.statusText}`;
-                  devResponse.text().then((err) => {
-                    config?.logger.error(err);
-                    res.end();
-                  });
-                }
-              })
-              .catch((error) => {
-                res.statusCode = 500;
-                res.statusMessage =
-                  `Unable to start dev mode in Foundry: ${error.message}`;
+            const settingsResponse = await setWidgetSettings(
```

Comment:
Is it worth logging what's happening? "Setting widget settings" "Enabling dev mode" in the vite dev mode output or is it not really shown

Reply:
It self-overwrites, and this only gets triggered when the user clicks the link to their browser so they won't actually see the output